kv/dist_sender.go

```diff
@@ -655,14 +655,17 @@ func (ds *DistSender) Send(_ context.Context, call proto.Call) {
 
 		var desc, descNext *proto.RangeDescriptor
 		err := retry.WithBackoff(retryOpts, func() (retry.Status, error) {
-			var descErr error
+			var err error
 			// Get range descriptor (or, when spanning range, descriptors).
 			// sendAttempt below may clear them on certain errors, so we
 			// refresh (likely from the cache) on every retry.
-			desc, descNext, descErr = ds.getDescriptors(call)
+			desc, descNext, err = ds.getDescriptors(call)
 			// If getDescriptors fails, we don't fish for retryable errors.
-			if descErr != nil {
-				return retry.Break, descErr
+			if err != nil {
+				if rErr, ok := err.(util.Retryable); ok && rErr != nil && rErr.CanRetry() {
```

Comment:
Sorry, that's not what I mean - the type assertion happens at runtime, and if the value is nil, it will not succeed (because there is no runtime type information) **i think**.